components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingPreferenceManagerImpl.java

```diff
@@ -702,4 +710,34 @@ private void validatePreferenceUrls(BrandingPreference preference) throws Brandi
             }
         }
     }
+
+    /**
+     * Clear the branding resolver cache if the branding preference enabled config is updated.
+     *
+     * @param previousBrandingPreference Previous branding preference.
+     * @param updatedBrandingPreference Updated branding preference.
+     * @param tenantDomain          Tenant domain.
+     * @throws BrandingPreferenceMgtException If an error occurs while clearing branding resolver cache hierarchy.
+     */
+    private void clearBrandingResolverCacheIfRequired(BrandingPreference previousBrandingPreference,
+                                                      BrandingPreference updatedBrandingPreference, String tenantDomain)
+            throws BrandingPreferenceMgtException {
+
+        JSONObject previousPreference = new JSONObject((LinkedHashMap) previousBrandingPreference.getPreference());
+        JSONObject updatedPreference = new JSONObject((LinkedHashMap) updatedBrandingPreference.getPreference());
+
+        // If configs.isBrandingEnabled is not found in preferences,  it is assumed that branding is enabled by default.
+        boolean isPreviousPreferencesPublished = !previousPreference.has("configs") ||
+                previousPreference.getJSONObject("configs").optBoolean("isBrandingEnabled", true);
+        boolean isUpdatedPreferencesPublished = !updatedPreference.has("configs") ||
+                updatedPreference.getJSONObject("configs").optBoolean("isBrandingEnabled", true);
+
+        if (isPreviousPreferencesPublished == isUpdatedPreferencesPublished) {
+            return;
+        }
+
+        BrandingPreferenceManagerComponentDataHolder.getInstance().getUiBrandingPreferenceResolver()
```

Comment:
Thought that  having a private method to return `BrandingPreferenceManagerComponentDataHolder.getInstance().getUiBrandingPreferenceResolver()` will make code more readable. WDYT?

Reply:
Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/41/commits/72cd2eb7d53698c795918b25058de2421d59dfbd